src/torsion_profile.py

```diff
@@ -0,0 +1,191 @@
+        super(TorsionProfileTarget, self).__init__(options, tgt_opts, forcefield)
+
+        if not os.path.exists(os.path.join(self.tgtdir, 'metadata.json')):
+            raise RuntimeError('TorsionProfileTarget needs torsion drive metadata.json file')
+        with open(os.path.join(self.tgtdir, 'metadata.json')) as f:
+            self.metadata = json.load(f)
+            self.ndim = len(self.metadata['torsion_grid_ids'][0])
+            self.freeze_atoms = sorted(list(set(itertools.chain(*self.metadata['dihedrals']))))
+        
+        ## Read in the coordinate files and get topology information from PDB
+        if hasattr(self, 'pdb') and self.pdb is not None:
+            self.mol = Molecule(os.path.join(self.root,self.tgtdir,self.coords),
+                                top=(os.path.join(self.root,self.tgtdir,self.pdb)))
+        else:
+            self.mol = Molecule(os.path.join(self.root,self.tgtdir,self.coords))
+        ## Number of snapshots.
+        self.ns = len(self.mol)
+        ## Option for how much data to write to disk.
+        self.set_option(tgt_opts,'writelevel','writelevel')
+        ## Harmonic restraint for non-torsion atoms in kcal/mol.
+        self.set_option(tgt_opts,'restrain_k','restrain_k')
+        ## Attenuate the weights as a function of energy
+        self.set_option(tgt_opts,'attenuate','attenuate')
+        ## Energy denominator for objective function
+        self.set_option(tgt_opts,'energy_denom','energy_denom')
+        ## Set upper cutoff energy
+        self.set_option(tgt_opts,'energy_upper','energy_upper')
+        ## Read in the reference data.
+        self.read_reference_data()
+        ## Build keyword dictionaries to pass to engine.
+        engine_args = OrderedDict(list(self.OptionDict.items()) + list(options.items()))
+        del engine_args['name']
+        ## Create engine object.
+        engine_args['freeze_atoms'] = self.freeze_atoms
+        self.engine = self.engine_(target=self, mol=self.mol, **engine_args)
+
+    def read_reference_data(self):
+
+        """ Read the reference ab initio data from a file such as qdata.txt.
+
+        After reading in the information from qdata.txt, it is converted
+        into the GROMACS energy units (kind of an arbitrary choice).
+        """
+        ## Reference (QM) energies
+        self.eqm           = []
+        ## The qdata.txt file that contains the QM energies and forces
+        self.qfnm = os.path.join(self.tgtdir,"qdata.txt")
+        # Parse the qdata.txt file
+        for line in open(os.path.join(self.root,self.qfnm)):
+            sline = line.split()
+            if len(sline) == 0: continue
+            elif sline[0] == 'ENERGY':
+                self.eqm.append(float(sline[1]))
+
+        if len(self.eqm) != self.ns:
+            raise RuntimeError("Length of qdata.txt should match number of structures")
+
+        # Turn everything into arrays, convert to kcal/mol
```

Comment:
I'm wondering if this is a good time to standardize the energy unit. Do we want to use `kcal/mol` everywhere, or is `kJ/mol` better?

Reply:
kJ/mol is kind of a legacy of abinitio.py which is a very old target, the first one.  I've since found that people tend to assume energies are in kcal/mol which is why I am going with it for torsion profiles.  We could start a new Github issue to discuss this, but I think that would quickly become a full blown thing folks will ask us to start requiring simtk.unit everywhere.  It could be worth considering but it's a lot of work.

Reply:
Yeah I agree sticking with one unit would good. If `kcal/mol` is the choice we can later make other target also use it. 👍